opencog/nlp/learn/scm/link-pipeline.scm

```diff
@@ -441,10 +441,19 @@
 			(set! start-time now)
 			(set! run-time run))))
 
-; ---------------------------------------------------------------------
-(define-public (observe-text plain-text)
+; --------------------------------------------------------------------
+(define-public (observe-text-mode plain-text observe-mode count-reach)
 "
  observe-text -- update word and word-pair counts by observing raw text.
+ 
+ There are currently two observing modes, set by observe-mode, both taking
+ an integer parameter:
+ - tree: counts pairs of words linked by the LG parser in 'any' language.
+ 	   'count-reach' specifies how many linkages from LG-parser to use.
```

Comment:
please change this to "any".  There are three different kinds of trees, today: any, ady and amy. Ady splits words into two morpheme (dual) and  amy splits words into multiple morphemes. Because of the way amy works, it might get redesigned in the future to work differently, so we have to leave room for yet other random languages.

In addition, there are theoretical reasons to explore non-tree random parses. These might be useful in handling finnish, turkish texts. Also, the poper handling of semitic texts (hebrew, arabic, ethioprian) is unclear.   That's why the language is called "any" and not "tree".

Reply:
also, in case its not clear, for the ady random tree language, exploring the clique would be incorrect.  You could make a weak argument that a clique-based approach to the morphology of finnish, turkish might be OK.  Maybe. But a properly tuned version of amy will outperform clique.  I've thought about this more heavily with Lithuanian, where there would be lots of clique-link links between morphemes, but only one link between words.  This would also be the case for the slavic languages.   From what I can tell, all of the romance languages are best handled by ady, and the clique language would be an utter disaster.   English is very special and unusual as a language.

Reply:
Sounds convincing ;)